File: ABCtask1/rectangle.cpp
#include <fstream>
#include "rectangle.h"

int Rectangle::In(FILE *input_file)
{
    char color_type[10001];
    int res = fscanf(input_file, "%d %d %d %d %10000s", &x_up_left, &y_up_left, &x_down_right, &y_down_right, color_type);
    if (res != 5)
    {
        std::cout<<x_up_left<<y_up_left<<std::endl;
        printf("Could not read string\n");
        return 1;
    }
    color = GetColor(color_type);
    return 0;
}

void Rectangle::Out(FILE *output_file)
{
    printf("Rectangle: (%d; %d); (%d; %d); Color = %s; \n Perimeter = %lf \n",
         x_up_left, y_up_left, x_down_right, y_down_right, ColorToString(color).c_str(), Perimeter());
    fprintf(output_file, "Rectangle: (%d; %d); (%d; %d); Color = %s; \n Perimeter = %lf \n",
            x_up_left, y_up_left, x_down_right, y_down_right, ColorToString(color).c_str(), Perimeter());
}

double Rectangle::Perimeter()
{
    return 2 * (abs(x_up_left - x_down_right) + abs(y_up_left - y_down_right));
}

std::string Rectangle::ColorToString(Color color)
{
    switch (color)
    {
    case RED:
        return "RED";
    case ORANGE:
        return "ORANGE";
    case YELLOW:
        return "YELLOW";
    case GREEN:
        return "GREEN";
    case LIGHT_BLUE:
        return "LIGHT_BLUE";
    case BLUE:
        return "BLUE";
    case VIOLET:
        return "VIOLET";
    default:
        return "NOT A COLOR!";
    }
}

Rectangle::Color Rectangle::GetColor(char color_type[10001])
{
    if (strcmp(color_type, "RED") == 0)
    {
        return RED;
    }
    if (strcmp(color_type, "ORANGE") == 0)
    {
        return ORANGE;
    }
    if (strcmp(color_type, "YELLOW") == 0)
    {
        return YELLOW;
    }
    if (strcmp(color_type, "GREEN") == 0)
    {
        return GREEN;
    }
    if (strcmp(color_type, "LIGHT_BLUE") == 0)
    {
        return LIGHT_BLUE;
    }
    if (strcmp(color_type, "BLUE") == 0)
    {
        return BLUE;
    }
    if (strcmp(color_type, "VIOLET") == 0)
    {
        return VIOLET;
    }
    return VIOLET;
}

File: pythonProject/rectangle.py
import random


# Class of rectangle
class Rectangle:
    def __init__(self):
        self.x_up_left = 0
        self.y_up_left = 0
        self.x_down_right = 0
        self.y_down_right = 0
        self.color = ""
        self.perimeter = 0.0

    # Inputting rectangle data from existing file
    def rectangle_in(self, params):
        if len(params) != 6:
            print("Could not read string")
            return 1
        else:
            self.x_up_left = int(params[1])
            self.y_up_left = int(params[2])
            self.x_down_right = int(params[3])
            self.y_down_right = int(params[4])
            self.color = params[5]
            self.perimeter = 2 * (abs(self.x_up_left - self.x_down_right)
                                  + abs(self.y_up_left - self.y_down_right))

    # Getting random parameters for rectangle
    def rectangle_in_rnd(self):
        while self.x_up_left >= self.x_down_right | self.y_up_left <= self.y_down_right:
            self.x_up_left = random.randint(1, 1000)
            self.y_up_left = random.randint(1, 1000)
            self.x_down_right = random.randint(1, 1000)
            self.y_down_right = random.randint(1, 1000)
        self.get_color()
        self.perimeter = 2 * (abs(self.x_up_left - self.x_down_right)
                                + abs(self.y_up_left - self.y_down_right))

    # Outputting rectangle data to console and output file
    def out(self, file):
        output = ("Rectangle: (" + str(self.x_up_left) + "; " + str(self.y_up_left) + "); (" + str(self.x_down_right)
                  + "; " + str(self.y_down_right) + "); Color = " + self.color + "Perimeter = " + str(self.perimeter))
        file.write(output)
        print(output)

    # Random generation for rectangle color
    def get_color(self):
        color_int = random.randint(0, 6)
        if color_int == 0:
            self.color = "RED\n"
        if color_int == 1:
            self.color = "ORANGE\n"
        if color_int == 2:
            self.color = "YELLOW\n"
        if color_int == 3:
            self.color = "GREEN\n"
        if color_int == 4:
            self.color = "LIGHT_BLUE\n"
        if color_int == 5:
            self.color = "BLUE\n"
        if color_int == 6:
            self.color = "VIOLET\n"
        return 0

